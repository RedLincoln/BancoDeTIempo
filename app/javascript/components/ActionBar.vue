<template>
  <v-toolbar flat>
    <template v-if="loggedIn">
      <v-btn
        v-if="role === 'admin'"
        data-testid="admin-users-tab"
        :to="{ name: 'admin-users' }"
        text
        >Gestionar Usuarios</v-btn
      >
    </template>
  </v-toolbar>
</template>

<script>
import { mapState } from "vuex";

export default {
  computed: mapState("session", ["loggedIn", "role"]),
};
</script>

<style>
</style>