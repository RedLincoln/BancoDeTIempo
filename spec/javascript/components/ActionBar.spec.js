import Vuetify from "vuetify";
import Vuex from "vuex";

import ActionBar from "../../../app/javascript/components/ActionBar.vue";

import { createLocalVue, mount } from "@vue/test-utils";

function getStoreByLogged({ loggedIn, role }) {
  const state = {
    user_name: "Michael",
    role,
    loggedIn,
  };
  return new Vuex.Store({
    modules: {
      session: {
        namespaced: true,
        state,
      },
    },
  });
}

describe("Navbar.vue", () => {
  const localVue = createLocalVue();
  let vuetify;
  let wrapper;

  beforeEach(() => {
    jest.resetAllMocks();
    localVue.use(Vuex);
  });

  it("renders properly", () => {
    vuetify = new Vuetify();
    wrapper = mount(ActionBar, {
      localVue,
      vuetify,
      store: getStoreByLogged({ loggedIn: true, role: "admin" }),
      stubs: ["router-link"],
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
