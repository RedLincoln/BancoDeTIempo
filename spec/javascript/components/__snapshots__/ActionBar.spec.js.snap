// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar.vue renders properly 1`] = `<div></div>`;
